### ISSUE TITLE
New Dataset

### ISSUE THREAD

**Issue by a897456** [drowe67/codec2#51]
I want to compress the test dataset of librispeech through codec2 to complete multiple bit rates, such 1.2kbps  2.4kbps 0.6kbps.  How can I achieve this?

**Comment by tmiw** [comment]
Are you using pytorch or TensorFlow to access librispeech, or some other mechanism? If one of the former two, you're probably using Python, so you'll need to hook into codec2 from there. I'd look into the cffi package (https://cffi.readthedocs.io/en/latest/overview.html) for generating a Python wrapper, or possibly [modifying the existing demo script](https://github.com/drowe67/codec2/blob/main/demo/freedv_700d_rx.py) to achieve what you want.

Additionally, the `c2enc` and `c2dec` programs in the tree are good to check out for determining the functions you need to call to perform the compression and decompression.

**Comment by a897456** [comment]
Pytorch and Python

**Comment by a897456** [comment]
1. `in.wav` to `in.raw`
2. (base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw hts1a_c2.bit
3. (base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2dec 2400 hts1a_c2.bit out.raw
4. `out.raw` to `out.wav`

`in.wav` is my new dataset, if I want to use codec2 with 2400bps ,is the command above correct? 
The `out.wav` is bad.
what is the `hts1a_c2.bit` ?

**Comment by tmiw** [comment]
Based on the source code for c2enc and c2dec, you should probably use file names ending in .c2 or .raw, i.e.

```
base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw in.c2
(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2dec 2400 in.c2 out.raw
```

But to answer your questions: yes, the command lines are correct and no, there's no specific meaning for `hts1a_c2.bit` (other than what c2enc and c2dec should write and read to/from, respectively).

**Comment by a897456** [comment]
THS，one more question:
`(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw in.c2`
Can 2400 be replaced by any other number?
If not, how should this parameter be set? By sampling rate or number of channels?

**Comment by a897456** [comment]
> Based on the source code for c2enc and c2dec, you should probably use file names ending in .c2 or .raw, i.e.

`in.bit` is ok?

> ```
> base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw in.c2
> (base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2dec 2400 in.c2 out.raw
> ```

`(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw in.bit`
`(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2dec 2400 in.bit out.raw`

**Comment by tmiw** [comment]
> Can 2400 be replaced by any other number? If not, how should this parameter be set? By sampling rate or number of channels?

There are only a certain set of values supported. From `c2enc.c`:

```c
  if (argc < 4) {
    printf(
        "usage: c2enc 3200|2400|1600|1400|1300|1200|700C "
        "InputRawspeechFile OutputBitFile [--natural] [--softdec] "
        "[--bitperchar] [--mlfeat f32File modelFile] [--loadcb stageNum "
        "Filename] [--var] [--eq]\n");
    printf("e.g. (headerless)    c2enc 1300 ../raw/hts1a.raw hts1a.bin\n");
    printf(
        "e.g. (with header to detect mode)   c2enc 1300 ../raw/hts1a.raw "
        "hts1a.c2\n");
    exit(1);
  }
```

The numbers generally do correspond to bit rate, though.

> `in.bit` is ok?
> 
> [...]
> 
> `(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2enc 2400 ../raw/in.raw in.bit` `(base) rian0532@DESKTOP-FF4J7IB:/mnt/e/00/codec2-main/build_linux$ ./src/c2dec 2400 in.bit out.raw`

I think any other extension is okay, but the advantage of the .c2 file format is that it will auto-detect the mode when decoding, which might reduce the number of things that could go wrong.
